sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala

```diff
@@ -259,6 +259,24 @@ class IncrementalExecution(
     }
   }
 
+  /**
+   * This rule populates the column family schemas for the TransformWithStateExec
+   * operator to ship them from the driver, where the schema and serializer objects
+   * are created, to the executor.
+   */
+  object StateStoreColumnFamilySchemasRule extends SparkPlanPartialRule {
+    override val rule: PartialFunction[SparkPlan, SparkPlan] = {
+      case statefulOp: StatefulOperator =>
+        statefulOp match {
+          case op: TransformWithStateExec =>
+            op.copy(
+              columnFamilySchemas = op.getColFamilySchemas()
+            )
```

Comment:
this is a bit confusing imho. If you have the `getColFamilySchemas` method as part of the class available, why do you have to set it on the class with a copy.

Two possible suggestions:
1. Make the `getColFamilySchemas` a static method. Not sure if that's possible though looking at the logic a bit more in TransformWithStateExec. It feels weird that you're opening and closing these handles just to get some of the information out.
2. Add a comment here that this needs to be run on the Driver, and also instead rename the method to: `withColumnFamilySchemas` which calls copy internally.